### ISSUE TITLE
[BUG] Install-AksHci stuck on "Waiting for pod 'Cloud Operator' to be ready..."

### ISSUE THREAD

**Issue by eponerine** [Azure/aks-hci#99]
**Describe the bug**
While running `Install-AksHci`, I am stuck on this step for over 60 minutes:

![image](https://user-images.githubusercontent.com/503974/119549297-136a1d00-bd65-11eb-83be-4f1a93f8a0b4.png)

Using `kubectl` with the working directory's kubeconfig file to peek inside the Control Plane node, I see the following error about **NetworkPluginNotReady**:

```
PS C:\Users\ernie\Downloads\1.0.0.10517> .\kubectl.exe describe nodes --kubeconfig=".\kubeconfig-mgmt"
Name:               moc-lxll7hja20n
Roles:              <none>
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=moc-lxll7hja20n
                    kubernetes.io/os=linux
Annotations:        volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Tue, 25 May 2021 14:15:00 -0400
Taints:             node.kubernetes.io/not-ready:NoSchedule
Unschedulable:      false
Conditions:
  Type             Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----             ------  -----------------                 ------------------                ------                       -------
  MemoryPressure   False   Tue, 25 May 2021 14:20:32 -0400   Tue, 25 May 2021 14:14:59 -0400   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure     False   Tue, 25 May 2021 14:20:32 -0400   Tue, 25 May 2021 14:14:59 -0400   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure      False   Tue, 25 May 2021 14:20:32 -0400   Tue, 25 May 2021 14:14:59 -0400   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready            False   Tue, 25 May 2021 14:20:32 -0400   Tue, 25 May 2021 14:14:59 -0400   KubeletNotReady              runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
Addresses:
  InternalIP:  172.20.91.21
  Hostname:    moc-lxll7hja20n
Capacity:
 cpu:                4
 ephemeral-storage:  100256840Ki
 hugepages-1Gi:      0
 hugepages-2Mi:      0
 memory:             7959788Ki
 pods:               110
Allocatable:
 cpu:                4
 ephemeral-storage:  92396703592
 hugepages-1Gi:      0
 hugepages-2Mi:      0
 memory:             7857388Ki
 pods:               110
System Info:
 Machine ID:                 0605d3eea165483ca80be3f88c5e4051
 System UUID:                af7d6fcc-2896-47a7-bd61-7c9a7938ac38
 Boot ID:                    c3094622-0889-4929-a81c-50b9b7b30f97
 Kernel Version:             5.10.28.1-4.cm1
 OS Image:                   CBL-Mariner/Linux
 Operating System:           linux
 Architecture:               amd64
 Container Runtime Version:  docker://19.3.11
 Kubelet Version:            v1.19.7
 Kube-Proxy Version:         v1.19.7
Non-terminated Pods:         (6 in total)
  Namespace                  Name                                       CPU Requests  CPU Limits  Memory Requests  Memory Limits  AGE
  ---------                  ----                                       ------------  ----------  ---------------  -------------  ---
  kube-system                etcd-moc-lxll7hja20n                       0 (0%)        0 (0%)      0 (0%)           0 (0%)         6m20s
  kube-system                kms-plugin-moc-lxll7hja20n                 500m (12%)    500m (12%)  128Mi (1%)       128Mi (1%)     7m36s
  kube-system                kube-apiserver-moc-lxll7hja20n             250m (6%)     0 (0%)      0 (0%)           0 (0%)         6m13s
  kube-system                kube-controller-manager-moc-lxll7hja20n    200m (5%)     0 (0%)      0 (0%)           0 (0%)         7m23s
  kube-system                kube-scheduler-moc-lxll7hja20n             100m (2%)     0 (0%)      0 (0%)           0 (0%)         6m19s
  kube-system                kube-vip-moc-lxll7hja20n                   0 (0%)        0 (0%)      0 (0%)           0 (0%)         6m25s
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests     Limits
  --------           --------     ------
  cpu                1050m (26%)  500m (12%)
  memory             128Mi (1%)   128Mi (1%)
  ephemeral-storage  0 (0%)       0 (0%)
```

- The ASHCI physical nodes are on `VLAN 140`, which is subnet `172.20.40.0/24`
- The k8s VMs will be on `VLAN 181`, which is subnet `172.20.80.0/20`... but specifically the VM and VIP range will be between `172.20.91.1 - 255`
- The DNS servers are on a different VLAN altogether, but routing works

I used the following config and then ran `Install-AksHci` to which the error above happens:

```
$vnet = New-AksHciNetworkSetting `
	-Name akshcivnet `
	-vSwitchName "LOGICAL SWITCH - Fully Converged (SDN)" `
	-k8sNodeIpPoolStart "172.20.91.21" `
	-k8sNodeIpPoolEnd "172.20.91.51" `
	-vipPoolStart "172.20.91.101" `
	-vipPoolEnd "172.20.91.199" `
	-ipAddressPrefix "172.20.80.0/20" `
	-gateway "172.20.80.1" `
	-dnsServers "172.20.25.62","172.20.25.64" `
	-vlanId 181
								 
								 
Set-AksHciConfig `
	-imageDir C:\ClusterStorage\AKS-01\Images `
	-workingDir C:\ClusterStorage\AKS-01\Working `
	-cloudConfigLocation C:\ClusterStorage\AKS-01\Config `
	-vnet $vnet `
	-cloudservicecidr "172.20.40.68/24" `
	-forceDnsReplication
```

I obviously can reach the `kubeapi` from `VLAN 140` (otherwise my `kubectl` test above wouldn't work). I can also hit `45000, 45001, 55000, and 65000` from `VLAN 181 -> 140` (tested with a dummy VM and `Test-NetConnection`)

**Environment:**
- OS: Azure Stack HCI build 10.0.17784.1705
- Browser: n/a
- Version: n/a
- AKS-HCI Version: GA 1.0.0.10517
- Kubernetes Version: 1.19.7 (I think?)

**Collect log files**
Logs available upon request; will not post publicly.

**Comment by madhanrm** [comment]
Currently the kvactl.exe thats waiting for the appliance to come up is loosing connectivity with the API server.
The root cause seems to be that kvactl.exe trying to connect to API Server periodically and ends up ending Syn Flood, which the firewall in the environment is catching and dropping, causing this process to hang and not complete the process.

Still looking into it to understand why this pattern of packets are being sent from kvactl

**Comment by madhanrm** [comment]
Still investigating and Looking to repro

**Comment by madhanrm** [comment]
@eponerine  can you please update the status here

**Comment by eponerine** [comment]
tl;dr - My issue wasn't SYN Flood, but instead [DNS resolution problem](https://content.spiceworksstatic.com/service.community/p/post_images/0000213914/582f18d2/attached_image/DNS.png). More details below. My k8s VMs tagged to the "Tenant VLAN" did not have their DNS servers configured to conditionally forward to my "Fabric VLAN" DNS servers, where the Cloud Agent stuff was registering to (separate VLANs and DNS/AD infrastructure). 

**Long Story -** 

After spending a few days w/ MSFT folks working on this, we've determined my specific problem seems to be _unrelated_ to the SYN flood. However, there def is room for improvements in `kvactl` to fix some lingering bugs :) 

As a reminder, in my scenario, I have 2x VLANs:

- The ASHCI physical nodes are on `VLAN 140`, which is subnet `172.20.40.0/24`
- The k8s VMs will be on `VLAN 181`, which is subnet `172.20.80.0/20`

And I have the following b-directional firewall rules configured between those VLANs:

- (TCP/UDP) 22, 53, 6443, 45000, 45001, 46000, 55000, 65000

When you do your `Install-AksHci`, you get a single VM that has a mini, self-contained k8s cluster on it (the VM name is `GUID-control-plane-0-#####`). As pods begin to spin up on this k8s cluster, one of them attempts to reach the Cloud Agent service on your ASHCI cluster (this is the resource called `ca-GUID`). It attempts to do this by FQDN, which means if your "Tenant VLAN" is unable to resolve FQDNs of things on your "Fabric VLAN", this will fail. 

In my case, I had port 53 TCP/UDP opened, **_but I did not have the DNS servers configured to conditionally forward to this other domain's name servers_**! D'oh!

As soon as I modified my DNS servers, I was able to get thru the entire install no problems. I have not confirmed with WAC yet, but I suspect that will succeed.

**Comment by madhanrm** [comment]
Can we close this issues please.
We have already fixed the bug that came out of this investigation. Thanks !
